skunkworks_crow/src/main/java/org/odk/share/services/SenderService.java

```diff
@@ -58,18 +58,54 @@ private void addUploadJobSubscription() {
                             break;
                     }
                 }).subscribe();
-
     }
 
-    public void startUploading(long[] instancesToSend, int port, int mode) {
+    /**
+     * build a bundle for transfer form data using hotspot.
+     *
+     * @param port            port for sending data via hotspot.
+     * @param instancesToSend instance to send.
+     * @param mode            review form mode or others.
+     */
+    private PersistableBundleCompat setupPersistableBundle(long[] instancesToSend, int port, int mode) {
         PersistableBundleCompat extras = new PersistableBundleCompat();
         extras.putLongArray(UploadJob.INSTANCES, instancesToSend);
+        extras.putBoolean("isBluetooth", false);
         extras.putInt(UploadJob.PORT, port);
         extras.putInt(MODE, mode);
+        return extras;
+    }
+
+    /**
+     * build a bundle for transfer form data using bluetooth.
+     *
+     * @param instancesToSend instance to send.
+     * @param mode            review form mode or others.
+     */
+    private PersistableBundleCompat setupPersistableBundle(long[] instancesToSend, int mode) {
+        PersistableBundleCompat extras = new PersistableBundleCompat();
+        extras.putLongArray(UploadJob.INSTANCES, instancesToSend);
+        extras.putBoolean("isBluetooth", true);
```

Comment:
Same here, use same string constant which will be created above and pass that to identify the mode of transfer.

Reply:
fixed.